feat(core): support dynamic click-through on bars based on configured modifier keys

VincentCassiau opened the pull request:
Adds the “click-through bars” functionality requested in Issue #130.

This pull request enhances the bar-click behaviour so that progress bars will only intercept mouse clicks when the user is holding a
modifier key that is actually configured for click-handling. In all other cases the click “passes through” to whatever UI lies beneath the bar.

VincentCassiau committed:
feat: dynamic click-through for bars based on configured modifier keys

funkydude commented:
This code would run continually even when not in a battleground, it would need to be changed.

VincentCassiau committed:
fix(core): only register MODIFIER_STATE_CHANGED in battlegrounds, arenas and test mode

- Add UpdateBarMouseState function to manage bar interactivity
- Register MODIFIER_STATE_CHANGED only in PvP zones
- Implement polling timer for test mode outside PvP zones
- Apply consistent implementation across all Core files (Retail, BCC, Vanilla, Wrath)
- Clean up timers and events properly in StopAllBars

VincentCassiau commented:
> This code would run continually even when not in a battleground, it would need to be changed.

Thank you ! I’ve made the change, I’ll test it a bit more thoroughly to make sure it’s the right modification and didn't break anything, and then I’ll push it afterwards !

VincentCassiau commented:
I've just pushed new changes to address two issues:

1. Added support for the "Test" mode that I had forgotten in my previous implementation
2. Fixed the code that was running continually even when not in a battleground, as requested

Please note that I don't have a way to test these changes on BCC Vanilla and Wrath versions. If needed, I can remove the modifications for these versions and keep only the retail changes. Let me know what you prefer, and I'll adjust accordingly !
